app/browser/basicAuth.js

```diff
@@ -0,0 +1,61 @@
+const electron = require('electron')
+const app = electron.app
+const appActions = require('../../js/actions/appActions')
+const appConstants = require('../../js/constants/appConstants')
+const appDispatcher = require('../../js/dispatcher/appDispatcher')
+const appStore = require('../../js/stores/appStore')
+
+// URLs to callback for auth.
+let authCallbacks = {}
+
+const cleanupAuthCallback = (tabId) => {
+  delete authCallbacks[tabId]
+}
+
+const runAuthCallback = (tabId, detail) => {
+  let cb = authCallbacks[tabId]
+  if (cb) {
+    delete authCallbacks[tabId]
+    if (detail) {
+      let username = detail.get('username')
+      let password = detail.get('password')
+      cb(username, password)
+    } else {
+      cb()
+    }
+  }
+}
+
+const doAction = (action) => {
```

Comment:
not sure how I feel about this. I wanted to separate the electron code from the appState to make it more unit testable, but not sure if this is the best way to handle it or not. I also considered some kind of change listener, but I think that might need a larger change to make sure the notifications are efficient